### ISSUE TITLE
在序列化时，超出指定的最大长度

### ISSUE THREAD

**Issue by yuaBing** [PaddlePaddle/PaddleNLP#2759]
欢迎您反馈PaddleNLP使用问题，非常感谢您对PaddleNLP的贡献！
在留下您的问题时，辛苦您同步提供如下信息：
- 版本、环境信息
1）PaddleNLP和PaddlePaddle版本：PaddleNLP 2.3.4，PaddlePaddle2.3.0
2）系统环境：官网的notobook ，python 3
- 复现信息：如为报错，请给出复现环境、复现步骤
- 
![image](https://user-images.githubusercontent.com/51101693/177786525-a77713e0-1dc1-4ce1-9cf9-a1d53d8c54b4.png)
如图，在tokenizer我指定了最大的长度为128，然后填充padding到最大长度max_length.
但最后我输出的时候，有大于128，如图。
![image](https://user-images.githubusercontent.com/51101693/177787101-b240ae90-7036-4912-85d5-8f45f5481ddd.png)

**Comment by w5688414** [comment]
请贴一下最小复现代码，方便我们复现和定位原因，不要截图哈

**Comment by yuaBing** [comment]
def convert_example(example,tokenizer,label_vocab,max_seq_len=512,is_test=False):
    #测试集没有标签
    if is_test:
        text = example
    else:
        text, label = example
    tokenizer_input = tokenizer.encode(text=text, max_length=128, padding='max_length', return_length=True, return_attention_mask=True)
    input_ids = tokenizer_input["input_ids"]
    token_type_ids = tokenizer_input["token_type_ids"]
    attention_masks = tokenizer_input["attention_mask"]
    seq_len = tokenizer_input["seq_len"]
    if not is_test:
        # 加入cls和sep
        if len(label) >= max_len - 2:
            label = label[0:max_len - 2]
        label = ['O']+label+['O']
        # 将标签转为序列
        label = [label_vocab[x] for x in label]
        if len(label) < max_len:
            label = label + [label_vocab['O']] * (max_len - len(label))
        return input_ids, token_type_ids, attention_masks, label, seq_len
    else: # 测试集，不返回标签
        return input_ids, token_type_ids, attention_masks, seq_len

tokenizer = RobertaTokenizer.from_pretrained('hfl/roberta-wwm-ext')
#偏函数，固定参数
trans_func = partial(convert_example, tokenizer=tokenizer, label_vocab=label_vocab, max_seq_len=128)
#对数据集进行编码
train_ds.map(trans_func)
dev_ds.map(trans_func)
print(train_ds[0])


#因为数据集train_ds、dev_ds的每条数据包含5部分，所以Tuple对象中包含5个批处理函数
batchify_fn = lambda samples, fn=Tuple(
    #将每条数据的input_ids组合为数组，如果input_ids不等长，那么填充为pad_val
    Pad(axis=0, pad_val=tokenizer.pad_token_id),
    # Stack(),
    #将每条数据的token_type_ids组合为数组，如果token_type_ids不等长，那么填充为pad_val
    Pad(axis=0, pad_val=tokenizer.pad_token_type_id),
    # Stack(),
    #attention_masks,补0
    Pad(axis=0, pad_val=0),
    #将每条数据的label组合为数组，等长
    Stack(),
    #将每条数据的seq_len组合为数组
    Stack(),  
): fn(samples)
#paddle.io.DataLoader加载给定数据集，返回迭代器，每次迭代访问batch_size条数据
#使用collate_fn定义所读取数据的格式
#训练集
train_loader = paddle.io.DataLoader(
    dataset=train_ds,
    batch_size=16,
    return_list=True,
    collate_fn=batchify_fn)

for step ,batch in enumerate(train_loader):
        input_ids,token_type_ids,masks,labels,seq_len= batch
        print("input_ids.shape:",input_ids.shape)
        print("token_type_ids.shape:",token_type_ids.shape)
        print("masks.shape:",masks.shape)
        print("labels.shape:",labels.shape)
        print("seq_len.shape:",seq_len.shape)
        if(step==5):
            break

**Comment by yuaBing** [comment]
数据集是用的resume

**Comment by w5688414** [comment]
```
from paddlenlp.transformers import RobertaTokenizer, RobertaModel

tokenizer = RobertaTokenizer.from_pretrained('hfl/roberta-wwm-ext')

text = '7月5日，深圳人大官网发布消息，《深圳经济特区智能网联汽车管理条例》（以下简称《条例》）已经深圳市人大常委会会议表决通过，自2022年8月1日起施行。这是国内首部关于智能网联汽车管理的法规，《条例》从道路测试和示范应用、准入和登记、使用管理、车路协同基础设施、网络安全和数据保护、交通违法和事故处理、法律责任等多个维度对智能网联汽车的管理进行全链条立法。'
tokenizer_input = tokenizer.encode(text=text, max_seq_len=128, return_length=True, return_attention_mask=True)
# max_seq_len
print(tokenizer_input)
print(len(tokenizer_input['attention_mask']))
```

这是输出：

```
{'input_ids': [101, 128, 3299, 126, 3189, 8024, 3918, 1766, 782, 1920, 2135, 5381, 1355, 2357, 3867, 2622, 8024, 517, 3918, 1766, 5307, 3845, 4294, 1277, 3255, 5543, 5381, 5468, 3749, 6756, 5052, 4415, 3340, 891, 518, 8020, 809, 678, 5042, 4917, 517, 3340, 891, 518, 8021, 2347, 5307, 3918, 1766, 2356, 782, 1920, 2382, 1999, 833, 833, 6379, 6134, 1104, 6858, 6814, 8024, 5632, 10550, 2399, 129, 3299, 122, 3189, 6629, 3177, 6121, 511, 6821, 3221, 1744, 1079, 7674, 6956, 1068, 754, 3255, 5543, 5381, 5468, 3749, 6756, 5052, 4415, 4638, 3791, 6226, 8024, 517, 3340, 891, 518, 794, 6887, 6662, 3844, 6407, 1469, 4850, 5745, 2418, 4500, 510, 1114, 1057, 1469, 4633, 6381, 510, 886, 4500, 5052, 4415, 510, 6756, 6662, 1291, 1398, 1825, 4794, 6392, 3177, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1], 'length': 128, 'seq_len': 128}
128
```
